### DISCUSSION TITLE
Close all sheets but the first one

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2514] [Q&A] [answered]
I'm trying to write a command that closes all the sheets except the first one. Here's my attempt:

```
@BaseSheet.api
def close_all_but_first(sheet: BaseSheet):
    for sheet in vd.sheets[1:]:
        vd.remove(sheet)
    vd.status(f'Closed all sheets except {vd.sheets[0]}')

BaseSheet.addCommand('gzq', 'close-all-but-first', 'close_all_but_first()')
```

This fails because `vd.sheets[0]` does not contain the first sheet. At least when I tried it contained the current sheet.

**How do I get the first sheet that was opened?**

**Comment by saulpw** [comment, marked as answer]
That's probably `vd.sheets[-1]`, or `vd.allSheets[0]`.

**Reply by cool-RR** [reply]
Thank you. Here's the corrected script for anyone who's interested:

```
@BaseSheet.api
def close_all_but_first(sheet: BaseSheet):
    for sheet in vd.allSheets[1:]:
        if sheet in vd.sheets:
            vd.remove(sheet)
    vd.status(f'Closed all sheets except {vd.allSheets[0]}')

BaseSheet.addCommand('gzq', 'close-all-but-first', 'close_all_but_first()')
```

I'm sure someone will be able to nitpick my inefficient solution :)
